Dockerfile

```diff
@@ -40,5 +38,12 @@ FROM base
 # Copy built application
 COPY --from=build /app /app
 
-# Start the server by default, this can be overwritten at runtime
-CMD [ "npm", "run", "start" ]
+# Set to `publish` or `api`
+# This argument controls what npm script is executed as the CMD.
+ARG SERVICE
+
+# ARGs are not preserved at runtime, we need to store the value
+# as a default value of an ENV var
+ENV SCRIPT="start:${SERVICE}"
+
+CMD npm run ${SCRIPT}
```

Comment:
@juliangruber do you have any opinion on the best Dockerfile & deployment configuration?

1. In the current proposal, we hard-code the service name at build time, but it can still be changed at run time via the `SCRIPT` env var
2. An alternative is to drop `ARG SERVICE` completely, let all services use the same Docker image, and change `fly.toml` config - remove `[build.args]` section and add `SCRIPT` env var config.
3. Another alternative is to use multi-stage build and add a new stage for each service. 

    ```Dockerfile
    FROM base as spark-api
    COPY --from=build /app /app
    CMD [ "npm", "run", "start:spark-api" ]
    
    FROM base as spark-publish
    COPY --from=build /app /app
    CMD [ "npm", "run", "start:spark-publish" ]
    ```

    We can specify the build target in the `fly.toml` file, see https://fly.io/docs/reference/configuration/#specify-a-multistage-docker-build-target

    ```toml
    [build]
      build-target = "spark-api"
    ```

Personally, I like the second option most. It is a bit simpler than the current `ARG SERVICE` variant.

Reply:
Tbh I don't have a preference here as all seems straight forward to me. Please choose what you prefer